### PR TITLE
Fix segfault when deleting the last history playlist under a year folder

### PR DESCRIPTION
**Pull request by daschuer**
This fixes https://bugs.launchpad.net/mixxx/+bug/1944029

The segfault reported here: https://github.com/mixxxdj/mixxx/pull/4297

### CODE REVIEW

File: src/library/trackset/baseplaylistfeature.cpp [hunk 1]
```diff
@@ -361,14 +361,16 @@ void BasePlaylistFeature::slotCreatePlaylist() {
...
-        QModelIndex nextIndex = start.sibling(i, start.column());
-        if (nextIndex.isValid()) {
-            TreeItem* pTreeItem = m_pSidebarModel->getItem(nextIndex);
-            DEBUG_ASSERT(pTreeItem != nullptr);
-            if (!pTreeItem->hasChildren()) {
-                return playlistIdFromIndex(nextIndex);
-            }
+    QModelIndex nextIndex = start.sibling(start.row() + 1, start.column());
+    if (!nextIndex.isValid() && start.row() > 0) {
+        // Now playlist below, look above
```
Comment (uklotzde) [inline src/library/trackset/baseplaylistfeature.cpp:366]: I don't understand this comment

Comment (uklotzde) [reply src/library/trackset/baseplaylistfeature.cpp:366]: What's the meaning of `start.row() > 0`?

File: src/library/trackset/baseplaylistfeature.cpp [hunk 2]
```diff
@@ -361,14 +361,16 @@ void BasePlaylistFeature::slotCreatePlaylist() {
...
+        nextIndex = start.sibling(start.row() + 1, start.column());
```
Comment (uklotzde) [inline src/library/trackset/baseplaylistfeature.cpp:367]: This expression will result in the same `nextIndex`!?

**Review by uklotzde — CHANGES_REQUESTED** [review]

**Comment by daschuer** [comment]
Ups, that was messed up. I have force pushed another try.

**Review by uklotzde — APPROVED** [review]
Thank you! LGTM

**Pull request merged**